tools/processresults.py

```diff
@@ -52,12 +62,52 @@ def ProcessResults(json_in_file, csv_prefix):
       # due to lack of data for a condition.
       print 'Exception in %s: %s' % (c, e.value)
 
+
 def _ParseSurveyResults(in_file):
   with open(in_file, 'r') as json_file:
     parsed = json.load(json_file)
+  demographic = filter(lambda x: x['survey_type'] == 'setup.js', parsed)
   events = filter(lambda x: x['survey_type'] != 'setup.js', parsed)
-  return events
+  return demographic, events
+  
+
+def _FilterDemographicResults(demo_res, discard_before_date):
+  """Return a list of results that occur after the given date.
+
+  Args:
+    parsed_demo: Results from demographic survey parsed from a raw JSON
+      file into list of dicts
+    discard_before_date: A date of type datetime.datetime; demographic
+      survey results before this date will be discarded
+    
+  Returns:
+    Integer index into the results list indicating which list
+    element's questions can be considered canonical and complete.
+    Also modifies the input results to filter out results with
+    PLACEHOLDER for every question or whose date_taken comes
+    before the given date.
+  """
+  demo_res[:] = _DiscardResultsBeforeDate(demo_res, discard_before_date)
```

Comment:
I was modifying the input, which may be bad practice anyway. I'll change it to return the modified results list instead of modifying the input.